### ISSUE TITLE
UI: Missing field separator after removing and re-adding field

### ISSUE THREAD

**Issue by zanderwar** [silverstripe/silverstripe-cms#2016]
![image](https://user-images.githubusercontent.com/13566916/32421589-1514f5ee-c2ff-11e7-8e5b-af1945ca62c4.png)

See between `EndDate` and `Club`

```php
class Event extends DataObject
{
    private static $db = [
        'Title'       => 'Varchar(255)',
        'Description' => 'Text',
        'StartDate'   => 'Datetime',
        'EndDate'     => 'Datetime',
    ];

    private static $many_many = [
        'Types' => EventType::class,
    ];

    private static $has_one = [
        'Location' => Location::class,
        'Club' => Club::class
    ];

    /**
     * @return \SilverStripe\Forms\FieldList
     */
    public function getCMSFields()
    {
        $fields = parent::getCMSFields();

        $fields->removeByName('Types');
        $fields->removeByName('LocationID');

        $fields->addFieldsToTab('Root.Main', [
            CheckboxSetField::create('Types', 'Types', EventType::get()),
            DropdownField::create('LocationID', 'Location', Location::getMappableRegions())
        ], 'StartDate');

        return $fields;
    }
}
```

**Comment by chillu** [comment]
Thanks Reece, can you please let us know which version this is in?

**Comment by zanderwar** [comment]
rc3 @chillu

**Comment by tractorcow** [comment]
Seems fixed in stable 4.0.0

![image](https://user-images.githubusercontent.com/936064/32863481-8a075324-cac0-11e7-9806-3413c54d3931.png)
